xbmc/filesystem/Directory.h

```diff
@@ -66,6 +68,14 @@ class CDirectory
                            , CFileItemList &items
                            , const CHints &hints);
 
+  typedef std::function<void(std::shared_ptr<CFileItem>& item)> DirectoryEnumerationCallback;
```

Comment:
We prefer  `using`over `typedef`

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#127-type-aliases